### PR TITLE
rnd 2867 metrics

### PR DESCRIPTION
**Pull request by arturk-vortexa**
Implementing metrics logging through the Pandera API. 

All of v4's metrics were migrated.

Migration scripts also implemented and tested locally.

All the user has to do is specify the desired metrics like this

image

### CODE REVIEW

File: tests/test_metrics.py [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+def test_metrics_logged_successfully():
+    test_path = TEST_RESOURCES / "data/input/parquet_sample.parquet"
+
+    resource = ParquetResource(path=test_path, pa_schema=ParquetSampleSchema)
+
+    with mock.patch("dynamicio.metrics.log_metric") as log_metric:
+        _ = resource.read()
+        log_metric.assert_has_calls(
+            [
+                call(column="foo_name", metric=Metric.COUNTS_PER_LABEL, value=8),
```
Comment (joloppo) [inline tests/test_metrics.py:41]: Why are there two of these?

**Review by joloppo — APPROVED** [review]
Nice work. Added a question about the test.

File: tests/test_metrics.py (hunk 1 above)
Comment (arturk-vortexa) [reply tests/test_metrics.py:41]: One for each label that it finds

**Comment by arturk-vortexa** [comment]
> Do existing tests pass?

They pass, and I've added a new test just for metrics!

**Pull request merged**